packages/xrpl/test/wallet/batchSigner.test.ts

```diff
@@ -0,0 +1,372 @@
+  algorithm: ECDSA.ed25519,
+})
+const otherWallet = Wallet.generate()
+
+const nonBatchTx = {
+  TransactionType: 'Payment',
+  Account: 'rJy554HmWFFJQGnRfZuoo8nV97XSMq77h7',
+  Destination: 'rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK',
+  Amount: '1000',
+}
+
+interface RawTransaction {
+  RawTransaction: BatchInnerTransaction
+}
+
+describe('Wallet batch operations', function () {
+  describe('signMultiBatch', function () {
+    let transaction: Batch
+
+    beforeEach(() => {
+      transaction = {
+        Account: 'rJCxK2hX9tDMzbnn3cg1GU2g19Kfmhzxkp',
+        Flags: 1,
+        RawTransactions: [
+          {
+            RawTransaction: {
+              Account: 'rJy554HmWFFJQGnRfZuoo8nV97XSMq77h7',
+              Flags: 0x40000000,
+              Amount: '5000000',
+              Destination: 'rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK',
+              Fee: '0',
+              Sequence: 215,
+              SigningPubKey: '',
+              TransactionType: 'Payment',
+            },
+          },
+          {
+            RawTransaction: {
+              Account: 'rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK',
+              Flags: 0x40000000,
+              Amount: '1000000',
+              Destination: 'rJCxK2hX9tDMzbnn3cg1GU2g19Kfmhzxkp',
+              Fee: '0',
+              Sequence: 470,
+              SigningPubKey: '',
+              TransactionType: 'Payment',
+            },
+          },
+        ],
+        TransactionType: 'Batch',
+      }
+    })
+    it('succeeds with secp256k1 seed', function () {
+      signMultiBatch(secpWallet, transaction)
+      const expected = [
+        {
+          BatchSigner: {
+            Account: 'rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK',
+            SigningPubKey:
+              '02691AC5AE1C4C333AE5DF8A93BDC495F0EEBFC6DB0DA7EB6EF808F3AFC006E3FE',
+            TxnSignature:
+              '304402207E8238D3D2B24B98BA925D69DDAFA3E7D07F85C8ABF1C040B3D1BEBE2C36E92B02200C122F7F3F86AB8FF89207539CAFB4613D665FF336796F99283ED94C66FB3094',
+          },
+        },
+      ]
+      assert.property(transaction, 'BatchSigners')
+      assert.strictEqual(
+        JSON.stringify(transaction.BatchSigners),
+        JSON.stringify(expected),
+      )
+    })
+
+    it('succeeds with ed25519 seed', function () {
+      signMultiBatch(edWallet, transaction)
+      const expected = [
+        {
+          BatchSigner: {
+            Account: 'rJy554HmWFFJQGnRfZuoo8nV97XSMq77h7',
+            SigningPubKey:
+              'ED3CC3D14FD80C213BC92A98AFE13A405A030F845EDCFD5E395286A6E9E62BA638',
+            TxnSignature:
+              '744FF09C11399F3AC1484F909A92F2D836EA979CB7655BC8F6BC3793F18892F92A16FE41C60EDCD6C2B757FF85D179F1589824ECA397EEA208B94C9D108CDF0A',
+          },
+        },
+      ]
+      assert.property(transaction, 'BatchSigners')
+      assert.strictEqual(
+        JSON.stringify(transaction.BatchSigners),
+        JSON.stringify(expected),
+      )
+    })
+
+    it('succeeds with a different account', function () {
```

Comment:
Can you help me understand the below three test cases. This is a regular key for which wallet? How do `regkeyWallet` and `batchAccount` play a role in the `expected` value? I'm not understanding the relationship between these accounts.

This design appears to be slightly different from that of xrpl-py Batch implementation.

Reply:
Same design, just a different test. This is ensuring that the wallets can properly handle a regular key wallet, where the public key differs from the account specified in the signing info.